domains/client/domain-executor/src/domain_block_processor.rs

```diff
@@ -182,6 +196,130 @@ where
         }
     }
 
+    pub(crate) fn fork_aware_block_imports(
```

Comment:
Would be helpful to have some comments on this function. It doesn't seem to import anything, so maybe a more descriptive name as well.

Reply:
Refactor it a bit, let me know if you have more questions or suggestions.